build-tools/packages/build-cli/src/commands/generate/assertTags.ts

```diff
@@ -97,71 +138,106 @@ export class TagAssertsCommand extends PackageCommand<typeof TagAssertsCommand>
 		}
 	}
 
-	protected async processPackage<TPkg extends Package>(
+	// This should not be called due to processPackages being overridden instead.
+	protected override async processPackage<TPkg extends Package>(
 		pkg: TPkg,
 		kind: PackageKind,
 	): Promise<void> {
-		const tsconfigPath = await this.getTsConfigPath(pkg);
-		this.collectAssertData(tsconfigPath);
+		throw new Error("Method not implemented.");
 	}
 
-	public async run(): Promise<void> {
-		// Calls processPackage on all packages to collect assert data.
-		await super.run();
+	protected override async processPackages(packages: PackageWithKind[]): Promise<string[]> {
+		const errors: string[] = [];
+
+		const collected: CollectedData = {
+			shortCodes: new Map<number, Node>(),
+			codeToMsgMap: new Map<string, string>(),
+			maxShortCode: -1,
+		};
+
+		const dataMap = new Map<PackageWithKind, PackageData>();
+
+		for (const pkg of packages) {
+			// Package configuration:
+			// eslint-disable-next-line no-await-in-loop
+			const tsconfigPath = await this.getTsConfigPath(pkg);
+			const packageConfig = getFlubConfig(pkg.directory).assertTagging;
+			const assertionFunctions: AssertionFunctions =
+				packageConfig?.assertionFunctions === undefined
+					? defaultAssertionFunctions
+					: new Map<string, number>(Object.entries(packageConfig.assertionFunctions));
+
+			// load the project based on the tsconfig
+			const project = new Project({
+				skipFileDependencyResolution: true,
+				tsConfigFilePath: tsconfigPath,
+			});
+
+			const newAssetFiles = this.collectAssertData(
+				project,
+				assertionFunctions,
+				collected,
+				errors,
+			);
+			dataMap.set(pkg, { assertionFunctions, newAssetFiles });
+		}
 
-		// Tag asserts based on earlier collected data.
-		this.tagAsserts(true);
-	}
+		if (errors.length > 0) {
+			return errors;
+		}
 
-	private collectAssertData(tsconfigPath: string): void {
-		// TODO: this can probably be removed now
```

Comment:
Removing this caused many more packages to be processed, like test-dds-utils, test-runtime-utils and everything in packages/test . See the config files added to filter them out.